op-batcher/batcher/channel_config_provider.go

```diff
@@ -14,7 +14,8 @@ const randomByteCalldataGas = params.TxDataNonZeroGasEIP2028
 
 type (
 	ChannelConfigProvider interface {
-		ChannelConfig() ChannelConfig
+		ChannelConfigFull() ChannelConfig
+		ChannelConfig(data []byte) ChannelConfig
```

Comment:
```suggestion
		ChannelConfig(frameDatas [][]byte) ChannelConfig
```
This method would take in a slice of frameDatas and then decide on the best way to submit them.

Reply:
The calculation does not need to be _perfect_, a best effort is already going to be good improvement.